src/test/java/brave/http/ITZipkinHttpClientImpl.java

```diff
@@ -63,6 +70,9 @@ void harnessSetup(Execution execution) {
 
     @Override protected void get(HttpClient client, String pathIncludingQuery) throws Exception {
         harness.yield(e -> client.get(URI.create(url(pathIncludingQuery)))).getValueOrThrow();
+        // Small delay to ensure the response interceptor is invoked before this method exits.
+        // This is needed for inherited test `redirect`
+        Thread.sleep(1);
```

Comment:
nit-picky but I generally find leveraging [Spock Polling or Blocking Conditions](http://spockframework.org/spock/javadoc/1.0/spock/util/concurrent/AsyncConditions.html) to make the tests more robust.  Though sometimes that takes some gymnastics to make happen.

Reply:
That makes a lot of sense.  These test use existing integration tests from the Brave project so hands are tied a bit.